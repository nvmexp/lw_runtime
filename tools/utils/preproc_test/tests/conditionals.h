#define DEFINED

 /**/ # \
     /**/ ifdef DEFINED
good
 /**/ # /**/ else
#bad bad
#endif

#ifndef DEFINED
#bad bad
#else
good
#endif

#if defined DEFINED
good
#else
#bad bad
#endif

#if defined(DEFINED)
good
#else
#bad bad
#endif

#if 1
good
#else
#bad bad
#endif

#ifdef UNDEFINED
#bad bad
#else
good
#endif

#ifndef UNDEFINED
good
#else
#bad bad
#endif

#if defined UNDEFINED
#bad bad
#else
good
#endif

#if defined(UNDEFINED)
#bad bad
#else
good
#endif

#if 0
#bad bad
#else
good
#endif

#if 1
good
#if 0
#bad bad
#include "non-existent"
#else
good
#endif
#else
#bad bad
#include "non-existent"
#endif

#pragma once
