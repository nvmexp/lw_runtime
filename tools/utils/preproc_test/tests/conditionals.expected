#line 1 "conditionals.h"




good







good



good





good





good







good



good







good





good





good



good




good







